test_host.cpp

```diff
@@ -208,7 +208,23 @@ void TestHost::EnsureFolderExists(const std::string &folder_path) {
   if (folder_path.length() > MAX_FILE_PATH_SIZE) {
     assert(!"Folder Path is too long.");
   }
-  if (!CreateDirectory(folder_path.c_str(), nullptr) && GetLastError() != ERROR_ALREADY_EXISTS) {
+
```

Comment:
The CreateDirectory() returns ERROR_PATH_NOT_FOUND when the intermediate paths are not available.  We could put this in a conditional to save the work for already calls to it where the intermediate paths already exist

Reply:
True, we could optimistically try to create the dir and then just do the whole thing recursively (I don't think it's worth walking backwards to the first missing and then having to walk forwards again).

In practice though, I doubt that the one or two extra `CreateDirectory` invocations that this function is likely to make will add up to a meaningful performance difference. Capturing the frame/z buffer and encoding them into PNG & writing them is almost certainly going to dominate the total time taken per test. I think for the sake of ease of maintenance we should just go with this unoptimized approach until it's clearly a bottleneck.

Reply:
👍